xdsl/dialects/snitch_runtime.py

```diff
@@ -334,10 +334,10 @@ class DmaStart2DBaseOperation(SnitchRuntimeBaseOperation, Generic[_T], ABC):
     T = Annotated[Attribute, ConstraintVar("T"), _T]
     dst: Operand = operand_def(_T)
     src: Operand = operand_def(_T)
-    dst_stride: Operand = operand_def(IndexType)
```

Comment:
Note to future selves: there may be other places where `IndexType` should be converted to `i32`/`i64`/etc.

Reply:
I would say that we might want to constrain it to the bitwidth of the snitch instructions as early as possible (in the dialects that are snitch-specific)